contribs/gnodev/cmd/gnobro/main.go

```diff
@@ -0,0 +1,452 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	charmlog "github.com/charmbracelet/log"
+	"github.com/charmbracelet/ssh"
+	"github.com/charmbracelet/wish"
+	"github.com/charmbracelet/wish/activeterm"
+	"github.com/charmbracelet/wish/bubbletea"
+	"github.com/charmbracelet/wish/logging"
+
+	"github.com/gnolang/gno/contribs/gnodev/pkg/browser"
+	"github.com/gnolang/gno/contribs/gnodev/pkg/events"
+	"github.com/gnolang/gno/gno.land/pkg/gnoclient"
+	"github.com/gnolang/gno/gno.land/pkg/integration"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/tm2/pkg/bft/rpc/client"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+const gnoPrefix = "gno.land"
+
+type broCfg struct {
+	readonly       bool
+	remote         string
+	devEndpoint    string
+	chainID        string
+	defaultAccount string
+	defaultRealm   string
+	sshListener    string
+	sshHostKeyPath string
+	banner         bool
+}
+
+var defaultBroOptions = broCfg{
+	remote:         "127.0.0.1:26657",
+	devEndpoint:    "",
+	sshListener:    "",
+	defaultRealm:   "gno.land/r/gnoland/home",
+	chainID:        "dev",
+	sshHostKeyPath: ".ssh/id_ed25519",
+}
+
+func main() {
+	cfg := &broCfg{}
+
+	stdio := commands.NewDefaultIO()
+	cmd := commands.NewCommand(
+		commands.Metadata{
+			Name:       "gnobro",
+			ShortUsage: "gnobro [flags] [pkg_path]",
+			ShortHelp:  "Gno Browser, a realm explorer",
+			LongHelp: `Gnobro is a terminal user interface (TUI) that allows you to browse realms within your
+terminal. It automatically connects to Gnodev for real-time development. In
+addition to hot reload, it also has the ability to execute commands and interact
+with your realm.
+`,
+		},
+		cfg,
+		func(_ context.Context, args []string) error {
+			return execBrowser(cfg, args, stdio)
+		})
+
+	cmd.Execute(context.Background(), os.Args[1:])
+}
+
+func (c *broCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.StringVar(
+		&c.remote,
+		"target",
+		defaultBroOptions.remote,
+		"target gnoland address",
+	)
+
+	fs.StringVar(
+		&c.chainID,
+		"chainid",
+		defaultBroOptions.chainID,
+		"chainid",
+	)
+
+	fs.StringVar(
+		&c.defaultAccount,
+		"account",
+		defaultBroOptions.defaultAccount,
+		"default local account to use",
+	)
+
+	fs.StringVar(
+		&c.defaultRealm,
+		"default-realm",
+		defaultBroOptions.defaultRealm,
+		"default realm to display when gnobro start and no argument are provided",
```

Comment:
Typo here, `starts`, and `arguments`

Reply:
@gfanton leftover 👀

Reply:
yes sorry d895eb0ca495252fe15ce8544c45c3174be2b9a8